apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -107,6 +110,23 @@ type ClusterIdentity struct {
 	ClusterName string `json:"clusterName"`
 }
 
+// UniqueName returns a user-friendly name that is unique *in the context of the current cluster*.
```

Comment:
Methods should be moved to the foreigncluster utils package, in the API we should only have types definitions

The method can have a signature like
```go
func UniqueName(*ClusterIdentity) string
```